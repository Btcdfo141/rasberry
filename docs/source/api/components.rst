:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
