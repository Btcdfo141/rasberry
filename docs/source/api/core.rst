.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: