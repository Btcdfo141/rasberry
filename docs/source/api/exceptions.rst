.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
