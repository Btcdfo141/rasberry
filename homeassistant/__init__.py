"""Init file for Home Assistant."""


test2
