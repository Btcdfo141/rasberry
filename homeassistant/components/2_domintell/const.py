"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
