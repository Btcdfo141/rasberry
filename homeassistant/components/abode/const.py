"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
