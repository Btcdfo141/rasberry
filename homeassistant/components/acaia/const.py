"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
