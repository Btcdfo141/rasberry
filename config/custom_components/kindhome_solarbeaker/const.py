from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
