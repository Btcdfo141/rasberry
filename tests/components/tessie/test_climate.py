"""Test the Tessie climate platform."""
from unittest.mock import patch

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.climate import (
    ATTR_HVAC_MODE,
    ATTR_PRESET_MODE,
    ATTR_TEMPERATURE,
    DOMAIN as CLIMATE_DOMAIN,
    SERVICE_SET_HVAC_MODE,
    SERVICE_SET_PRESET_MODE,
    SERVICE_SET_TEMPERATURE,
    SERVICE_TURN_ON,
    HVACMode,
)
from homeassistant.components.tessie.const import TessieClimateKeeper
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError

from .common import ERROR_UNKNOWN, TEST_RESPONSE, setup_platform


async def test_climate(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
    """Tests that the climate entity is correct."""

    assert len(hass.states.async_all(CLIMATE_DOMAIN)) == 0

    await setup_platform(hass)

    assert hass.states.async_all(CLIMATE_DOMAIN) == snapshot(name="all")

    entity_id = "climate.test_climate"

    # Test setting climate on
    with patch(
        "homeassistant.components.tessie.climate.start_climate_preconditioning",
        return_value=TEST_RESPONSE,
    ) as mock_set:
        await hass.services.async_call(
            CLIMATE_DOMAIN,
            SERVICE_SET_HVAC_MODE,
            {ATTR_ENTITY_ID: [entity_id], ATTR_HVAC_MODE: HVACMode.HEAT_COOL},
            blocking=True,
        )
        mock_set.assert_called_once()
    assert hass.states.get(entity_id) == snapshot(name=f"{entity_id}:on")

    # Test setting climate temp
    with patch(
        "homeassistant.components.tessie.climate.set_temperature",
        return_value=TEST_RESPONSE,
    ) as mock_set:
        await hass.services.async_call(
            CLIMATE_DOMAIN,
            SERVICE_SET_TEMPERATURE,
            {ATTR_ENTITY_ID: [entity_id], ATTR_TEMPERATURE: 20},
            blocking=True,
        )
        mock_set.assert_called_once()
    assert hass.states.get(entity_id) == snapshot(name=f"{entity_id}:temperature")

    # Test setting climate preset
    with patch(
        "homeassistant.components.tessie.climate.set_climate_keeper_mode",
        return_value=TEST_RESPONSE,
    ) as mock_set:
        await hass.services.async_call(
            CLIMATE_DOMAIN,
            SERVICE_SET_PRESET_MODE,
            {ATTR_ENTITY_ID: [entity_id], ATTR_PRESET_MODE: TessieClimateKeeper.ON},
            blocking=True,
        )
        mock_set.assert_called_once()
    assert hass.states.get(entity_id) == snapshot(name=f"{entity_id}:preset")

    # Test setting climate off
    with patch(
        "homeassistant.components.tessie.climate.stop_climate",
        return_value=TEST_RESPONSE,
    ) as mock_set:
        await hass.services.async_call(
            CLIMATE_DOMAIN,
            SERVICE_SET_HVAC_MODE,
            {ATTR_ENTITY_ID: [entity_id], ATTR_HVAC_MODE: HVACMode.OFF},
            blocking=True,
        )
        mock_set.assert_called_once()
    assert hass.states.get(entity_id) == snapshot(name=f"{entity_id}:off")


async def test_errors(hass: HomeAssistant) -> None:
    """Tests errors are handled."""

    await setup_platform(hass)
    entity_id = "climate.test_climate"

    # Test setting climate on with unknown error
    with patch(
        "homeassistant.components.tessie.climate.start_climate_preconditioning",
        side_effect=ERROR_UNKNOWN,
    ) as mock_set, pytest.raises(HomeAssistantError) as error:
        await hass.services.async_call(
            CLIMATE_DOMAIN,
            SERVICE_TURN_ON,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )
        mock_set.assert_called_once()
        assert error.from_exception == ERROR_UNKNOWN
