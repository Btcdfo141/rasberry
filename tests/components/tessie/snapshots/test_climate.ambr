# serializer version: 1
# name: test_climate[all]
  list([
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'current_temperature': 30.5,
        'friendly_name': 'Test Climate',
        'hvac_modes': list([
          <HVACMode.HEAT_COOL: 'heat_cool'>,
          <HVACMode.OFF: 'off'>,
        ]),
        'max_temp': 28.0,
        'min_temp': 15.0,
        'preset_mode': 'off',
        'preset_modes': list([
          <TessieClimateKeeper.OFF: 'off'>,
          <TessieClimateKeeper.ON: 'on'>,
          <TessieClimateKeeper.DOG: 'dog'>,
          <TessieClimateKeeper.CAMP: 'camp'>,
        ]),
        'supported_features': <ClimateEntityFeature: 17>,
        'temperature': 22.5,
      }),
      'context': <ANY>,
      'entity_id': 'climate.test_climate',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
  ])
# ---
# name: test_climate[set-preset]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 30.5,
      'friendly_name': 'Test Climate',
      'hvac_modes': list([
        <HVACMode.HEAT_COOL: 'heat_cool'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 28.0,
      'min_temp': 15.0,
      'preset_mode': <TessieClimateKeeper.ON: 'on'>,
      'preset_modes': list([
        <TessieClimateKeeper.OFF: 'off'>,
        <TessieClimateKeeper.ON: 'on'>,
        <TessieClimateKeeper.DOG: 'dog'>,
        <TessieClimateKeeper.CAMP: 'camp'>,
      ]),
      'supported_features': <ClimateEntityFeature: 17>,
      'temperature': 20.0,
    }),
    'context': <ANY>,
    'entity_id': 'climate.test_climate',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat_cool',
  })
# ---
# name: test_climate[set-temp]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 30.5,
      'friendly_name': 'Test Climate',
      'hvac_modes': list([
        <HVACMode.HEAT_COOL: 'heat_cool'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 28.0,
      'min_temp': 15.0,
      'preset_mode': 'off',
      'preset_modes': list([
        <TessieClimateKeeper.OFF: 'off'>,
        <TessieClimateKeeper.ON: 'on'>,
        <TessieClimateKeeper.DOG: 'dog'>,
        <TessieClimateKeeper.CAMP: 'camp'>,
      ]),
      'supported_features': <ClimateEntityFeature: 17>,
      'temperature': 20.0,
    }),
    'context': <ANY>,
    'entity_id': 'climate.test_climate',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat_cool',
  })
# ---
# name: test_climate[turn-off]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 30.5,
      'friendly_name': 'Test Climate',
      'hvac_modes': list([
        <HVACMode.HEAT_COOL: 'heat_cool'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 28.0,
      'min_temp': 15.0,
      'preset_mode': 'off',
      'preset_modes': list([
        <TessieClimateKeeper.OFF: 'off'>,
        <TessieClimateKeeper.ON: 'on'>,
        <TessieClimateKeeper.DOG: 'dog'>,
        <TessieClimateKeeper.CAMP: 'camp'>,
      ]),
      'supported_features': <ClimateEntityFeature: 17>,
      'temperature': 20.0,
    }),
    'context': <ANY>,
    'entity_id': 'climate.test_climate',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_climate[turn-on]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 30.5,
      'friendly_name': 'Test Climate',
      'hvac_modes': list([
        <HVACMode.HEAT_COOL: 'heat_cool'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 28.0,
      'min_temp': 15.0,
      'preset_mode': 'off',
      'preset_modes': list([
        <TessieClimateKeeper.OFF: 'off'>,
        <TessieClimateKeeper.ON: 'on'>,
        <TessieClimateKeeper.DOG: 'dog'>,
        <TessieClimateKeeper.CAMP: 'camp'>,
      ]),
      'supported_features': <ClimateEntityFeature: 17>,
      'temperature': 22.5,
    }),
    'context': <ANY>,
    'entity_id': 'climate.test_climate',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat_cool',
  })
# ---
